src/RewardsStreamerMP.sol

```diff
@@ -73,7 +96,91 @@ contract RewardsStreamerMP is UUPSUpgradeable, IStakeManager, TrustedCodehashAcc
+     */
+    function isVaultRegistered(address vault) public view returns (bool) {
+        return stakeVaultRegistry.vaultOwners(vault) != address(0);
+    }
+
+    /**
+     * @notice Registers a vault with its owner. Called by the vault itself during initialization.
+     * @dev Only callable by contracts with trusted codehash
+     */
+    function registerVault() external onlyTrustedCodehash {
+        address vault = msg.sender;
+        address owner = IStakeVault(vault).owner();
+        stakeVaultRegistry.register(owner, vault);
+    }
+
+    /**
+     * @notice Get all vaults owned by an address
+     */
+    function getUserVaults(address owner) external view returns (address[] memory) {
+        return stakeVaultRegistry.vaultsOf(owner);
+    }
+
+    /**
+     * @notice Get the total multiplier points for a user
+     * @dev Iterates over all vaults owned by the user and sums the multiplier points
+     * @param user The address of the user
+     * @return The total multiplier points for the user
+     */
+    function getUserTotalMP(address user) external view returns (uint256) {
+        address[] memory userVaults = stakeVaultRegistry.vaultsOf(user);
+        uint256 userTotalMP = 0;
+
+        for (uint256 i = 0; i < userVaults.length; i++) {
+            Account storage account = accounts[userVaults[i]];
+            userTotalMP += account.accountMP + _getAccruedMP(account);
+        }
+        return userTotalMP;
+    }
+
+    /**
+     * @notice Get the total maximum multiplier points for a user
+     * @dev Iterates over all vaults owned by the user and sums the maximum multiplier points
+     * @param user The address of the user
+     * @return The total maximum multiplier points for the user
+     */
+    function getUserTotalMaxMP(address user) external view returns (uint256) {
+        address[] memory userVaults = stakeVaultRegistry.vaultsOf(user);
+        uint256 userTotalMaxMP = 0;
+
+        for (uint256 i = 0; i < userVaults.length; i++) {
+            userTotalMaxMP += accounts[userVaults[i]].maxMP;
+        }
+        return userTotalMaxMP;
+    }
+
+    /**
+     * @notice Get the total staked balance for a user
+     * @dev Iterates over all vaults owned by the user and sums the staked balances
+     * @param user The address of the user
+     * @return The total staked balance for the user
+     */
+    function getUserTotalStakedBalance(address user) external view returns (uint256) {
+        address[] memory userVaults = stakeVaultRegistry.vaultsOf(user);
+        uint256 userTotalStake = 0;
+
+        for (uint256 i = 0; i < userVaults.length; i++) {
+            userTotalStake += accounts[userVaults[i]].stakedBalance;
+        }
+        return userTotalStake;
+    }
+
+    function stake(
+        uint256 amount,
+        uint256 lockPeriod
+    )
+        external
+        onlyTrustedCodehash
+        onlyNotEmergencyMode
+        onlyRegisteredVault
```

Comment:
Why is a requirement to have the vault registered? Is there any security benefits on that? For me it seems is just a tool. This type of verification can be done in UI, and there should be no problem in using StakeManger without register.

Reply:
I'd say, it's an enforced invariant that only registered vaults can stake.
If it's not enforced here, then any vault could stake and the vault might not be registered, meaning it would not be taken into consideration when querying the XP for an owner of vaults.

Reply:
Ok, but it does not pose any risk for the current system, this should be a requirement by the system which requires this. 
Otherwise the dependency graph is weird.

Reply:
> and there should be no problem in using StakeManger without register.

So, to clarify here: the reason we've introduced registration of vaults by owners, is so that we can aggregate all MP/XP/stakedBalance for a given owner.

If a StakeVault is not registered, it won't be known in the aggregation of XP.

Sure, we could say: Well, if a user ignores the rules and still stakes without registering, then it's their problem.

I think though, we shouldn't even allow that.

Reply:
I agree with keeping the registration of the vaults, it feels more consistent like this